testsuite/gnat2goto/tests/pragma_machine_attribute/pragma_machine_attribute.adb

```diff
@@ -0,0 +1,23 @@
+procedure Pragma_Machine_Attribute is
+
+   procedure Timer_Interrupt;
+   pragma Machine_Attribute (Timer_Interrupt, "signal");
+
+   Seconds_Count : Integer := 0;
+   Interval : Integer := 1;
+   Timeout : Boolean := False;
+
+   procedure Timer_Interrupt is
+   begin
+      Seconds_Count := Seconds_Count + 1;
+      if Seconds_Count = Interval then
+         Timeout := True;
+      end if;
+   end Timer_Interrupt;
+
+begin
+   Timer_Interrupt;
+
+   pragma Assert (Timeout and then Interval = Seconds_Count);
```

Comment:
FWIW we should really use `and` instead of `and then` when we don't explicitly need short-circuiting behaviour.

Reply:
Done.